pulsar-broker/src/main/java/org/apache/pulsar/compaction/CompactedTopicImpl.java

```diff
@@ -304,8 +306,10 @@ static CompletableFuture<List<Entry>> readEntries(LedgerHandle lh, long from, lo
      * Getter for CompactedTopicContext.
      * @return CompactedTopicContext
      */
-    public Optional<CompactedTopicContext> getCompactedTopicContext() throws ExecutionException, InterruptedException {
-        return compactedTopicContext == null ? Optional.empty() : Optional.of(compactedTopicContext.get());
+    public Optional<CompactedTopicContext> getCompactedTopicContext() throws ExecutionException, InterruptedException,
```

Comment:
Why do we fix here?

Reply:
See the answer here https://github.com/apache/pulsar/pull/23258/files#r1749751058